Script and sample data for leaderboard

bhaktipriya opened the pull request:
The Python script evaluates users based on the number and difficulty of issues solved and on the basis of additions, deletions and commits.

All data is in JSON format.

bhaktipriya committed:
The Algorithm and reasons for chosing it.

bhaktipriya committed:
TODO for osdg ranker

bhaktipriya committed:
Update The IDEA

bhaktipriya committed:
Merge pull request #1 from bhaktipriya/bhaktipriya-patch-1

Update The IDEA

bhaktipriya committed:
Labelled Ranker

Rankings based on labelled issues and pull request

bhaktipriya committed:
Merge branch 'ranker'

bhaktipriya committed:
Merged Ranker

bhaktipriya committed:
Removed files

bhaktipriya committed:
Removed token

bhaktipriya committed:
Added extra features

In case a repo is not labelled,
User will be rated on the basis additions,deletions and commits

bhaktipriya committed:
Merge branch 'master' of https://github.com/bhaktipriya/osdg-leaderboard
Added extra features
In case a repo is not labelled,
	    User will be rated on the basis additions,deletions and
	    commits

Conflicts:
	ranker/labelled_ranker.py

bhaktipriya committed:
Added extra features

 
    In case a repo is not labelled,
    User will be rated on the basis additions,deletions and commits

bhaktipriya committed:
Final updates

bhaktipriya committed:
Merge branch 'master' of https://github.com/bhaktipriya/osdg-leaderboard

bhaktipriya committed:
Updated script to sort JSON data

bhaktipriya committed:
Created a Responsive Bar chart for Leaderboard

bhaktipriya committed:
Typo correction

pmlhotra commented on TODO:
```diff
@@ -0,0 +1,23 @@
+
+
+				
+				
```
Not required anymore.

pmlhotra commented on The IDEA:
```diff
@@ -0,0 +1,32 @@
+  - Top Repos
+  - Top Bug Reporters
+
+We can always merge 1 and 3, but I feel keeping a separate category will attract more contributors.
```
Rename the file, no white spaces.

pmlhotra commented on ranker/labelled_ranker.py:
```diff
@@ -0,0 +1,118 @@
+import json, requests
```
Each import on a separate line.

pmlhotra commented on ranker/labelled_ranker.py:
```diff
@@ -0,0 +1,118 @@
+
+def filter_issues(issue_list):
+    """Returns a list of issues which have specific labels"""
+
```
Make it consistent, if you are having a new line after doc string then make it for all, otherwise just don't add it. When in doubts always use `pep8`.

pmlhotra commented on ranker/show/show_files/ab.html:
```diff
@@ -0,0 +1,135 @@
+			}
+			});
+		return json;
+		})(); 
```
Indentation gone awry.

pmlhotra commented on ranker/show/show_files/ab.html:
```diff
@@ -0,0 +1,135 @@
+
+
+
+
```
Why so many blank lines ?

pmlhotra commented on ranker/show/show_files/try.html:
```diff
@@ -0,0 +1,79 @@
+											        });
+			    return json;
+			    })(); 
+
```
Haha, why are they flying ?

pmlhotra commented on ranker/show/show_files/try.html:
```diff
@@ -0,0 +1,79 @@
+			        add.push([json_data [i]["scores"]["additions"]]);
+				    com.push([json_data [i]["scores"]["commits"]]);
+				        users.push([json_data [i]["user"]]);
+
```
These too.

pmlhotra commented on ranker/show/show_files/try.html:
```diff
@@ -0,0 +1,79 @@
+
+
+
+
```
Blank lines.

bitgeeky commented:
Nice Work ! and Thanks for the pull. Very good start, Let me know when you make the above mentioned changes so that it can be reviewed again. Also remove the swap files.

pmlhotra commented on ranker/labelled_ranker.py:
```diff
@@ -0,0 +1,118 @@
+
+def req(get,params={}):
+    """Function to send HTTP Requests"""
+    params['access_token']='d0fec873c0a68b8e8b29b2403914a41f59479dc5'
```
Making Access Tokens public is never a good idea. Use an environment variable or pass a command line argument.

pmlhotra commented on ranker/labelled_ranker.py:
```diff
@@ -0,0 +1,118 @@
+
+
+if __name__ == "__main__":
+    org="OSDG-IIITH"
```
Use an environment variable or pass as an argument so that the project can be used in general by anyone.

pmlhotra commented on ranker/labelled_ranker.py:
```diff
@@ -0,0 +1,118 @@
+    for i in range(len(users)):
+        users[i]={'user':users[i][0],'scores':users[i][1]}
+
+    with open(filename, "w") as jsonFile:
```
File reading and writing should be in a `try` `catch` block.

pmlhotra commented on ranker/show/show.html:
```diff
@@ -0,0 +1,53 @@
+		<meta http-equiv="edit-Type" edit="text/html; charset=utf-8">
+
+		
+			<script type="text/javascript" src="show_files/moo-clientcide-1.js"></script>
```
Use a good text editor.

pmlhotra commented on ranker/show/show.html:
```diff
@@ -0,0 +1,53 @@
+	    <script src="show_files/ga.js" type="text/javascript"></script>
+	    	
+
+</body></html>
```
Rename directories to variables that reflect distinction between frontend and backend of the application.

pmlhotra commented on ranker/show/show_files/ab.html:
```diff
@@ -0,0 +1,135 @@
+var bugs=[],easy=[],med=[],hard=[],del=[],add=[],com=[],users=[];
+
+for(var i in json_data)
+{
```
Remember, there are two types of people, and you can't be both. 

```
for(){
}
```

or

```
for()
{
}
```

pmlhotra commented on ranker/show/show_files/ab.html:
```diff
@@ -0,0 +1,135 @@
+type: 'column'
+},
+title: {
+text: 'OSDG-IIITH Leaderboard'
```
Its not Leaderboard, we agreed on this. Right ?

pmlhotra commented on ranker/labelled_ranker.py:
```diff
@@ -0,0 +1,118 @@
+
+if __name__ == "__main__":
+    org="OSDG-IIITH"
+    filename="show/show_files/ranking.json"
```
What's the difference between `show/show_files/ranking.json` and `show/ranking.json` ?

pmlhotra commented:
Nice Work ! and Thanks for the pull. Very good start, Let me know when you make the above mentioned changes so that it can be reviewed again. Also remove the swap files.

bhaktipriya committed:
Delete TODO

bhaktipriya committed:
Update and Rename The Idea to Concept

bhaktipriya committed:
Delete ranking.json

Duplicate copy. The file is already stored in show/show_files/ranking.json

bhaktipriya committed:
Prettyprinted ab.html and changed titles

bhaktipriya committed:
Prettyprinted show.html

bhaktipriya committed:
Formatted labelled_ranker.py

The script now follows the PEP8 conventions.

bhaktipriya committed:
Organised the code into folders

New folders reflect distinction between frontend and backend of the application.
The patch also involves refactoring of paths.

bhaktipriya committed:
Merge branch 'master' of https://github.com/bhaktipriya/osdg-leaderboard

Conflicts:
	ranker/a.html

bhaktipriya committed:
Added Sample Screenshots

bhaktipriya commented:
@bitgeeky . Thank you for your review. I have worked on the changes as pointed out by you.
Please let  me know if there are any more changes.

mrinaldhar commented:
Good start. Merging (finally!)

mrinaldhar commented:
I'd like to go over the need for a backend python script to handle that stuff? @bhaktipriya
